loading.bs

```diff
@@ -0,0 +1,949 @@
+spec: draft-thomson-http-mice; urlPrefix: https://tools.ietf.org/html/draft-thomson-http-mice-02#
+    type: dfn
+        text: mi-sha256 parameter; url: section-3.1
+        text: integrity proof for the first record; url: section-2.2
+spec: draft-yasskin-wpack-bundled-exchanges; urlPrefix: https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#
+    type: dfn
+        text: parsing a CBOR item; url: parse-known-length
+spec: draft-yasskin-http-origin-signed-responses; urlPrefix: https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#
+    type: dfn
+        text: stateful request header; url: stateful-headers
+        text: stateful response header; url: stateful-headers
+spec: draft-yasskin-httpbis-origin-signed-exchanges-impl-01; urlPrefix: https://tools.ietf.org/html/draft-yasskin-httpbis-origin-signed-exchanges-impl-01#
+    type: dfn
+        text: canonically-encoded CBOR; url: section-3.4
+        text: CanSignHttpExchanges; url: section-4.2
+        text: cert-chain CDDL; url: section-3.3
+        text: Signature header; url: section-3.1
+spec: draft-ietf-httpbis-header-structure; urlPrefix: https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-07#
+    text: Parsing HTTP1 Header Fields into Structured Headers; type: dfn; url: section-4.2
+spec: SHA2; urlPrefix: http://csrc.nist.gov/publications/fips/fips180-4/fips-180-4.pdf
+    type: dfn
+        text: SHA-256; url: #
+</pre>
+<pre class='link-defaults'>
+spec:fetch; type:dfn; for:/; text:response
+spec:streams; type:interface; text:ReadableStream
+</pre>
+
+# Introduction # {#intro}
+
+<em>This section is non-normative.</em>
+
+The Signed Exchanges specification
+[[draft-yasskin-http-origin-signed-responses]] describes a way to provide one or
+more signatures for an HTTP exchange and to check whether any of those
+signatures is trusted as authoritative for a particular origin. This
+specification describes how web browsers load those exchanges. It is expressed
+as several monkeypatches to the [[FETCH]] specification which call algorithms
+defined here.
+
+## Overview ## {#overview}
+
+When fetching a resource (`https://distributor.example.org/foo.sxg`) with the
+`application/signed-exchange` [=MIME type=], the UA parses it, checks its
+signatures, and then if all is well, redirects to its request URL
+(`https://publisher.example.org/foo`) with a "stashed" exchange attached to the
+request. The redirect applies all the usual processing, and then when it would
+normally check for an HTTP cache hit, it also checks whether the stashed request
+matches the redirected request and which of the stashed exchange or HTTP cache
+contents is newer. If the stashed exchange matches and is newer, the UA returns
+the stashed response.
+
+A Service Worker for `https://distributor.example.org/` gets to handle the
+original request. A Service Worker for `https://publisher.example.org/` does not
```

Comment:
@kinu Have y'all looked into the Service Worker side? I believe Fetch's https://fetch.spec.whatwg.org/#concept-http-fetch step 4.2

> If request’s redirect mode is "follow", then set request’s service-workers mode to "none".

prevents all redirects from hitting a service worker, so even if distributor.example.org just served a 302 to publisher.example.org, publisher.example.org's SW wouldn't get to handle the navigation. Am I misreading anything?

Reply:
For navigation the default redirect mode is "manual", therefore every time a request's redirected corresponding SW has a chance to see it.

Reply:
Aha! Thanks! 

I think both @cramforce's idea to use the `preloadResponse` and asking people to call `fetch()` with cache=="only-if-cached" will work, but to use "only-if-cached" we'll have to tell people to modify the original request or its clone, instead of creating a new request with the same URL. A new request with the same URL won't have the stashed exchange attached, and wiring those up will require an extra global map in the SW itself, which could also get confused by interleaved requests to different signed exchanges holding the same publication URL.